src/source/Common/AwsV4Signer.c

```diff
@@ -117,6 +117,12 @@ STATUS signAwsRequestInfo(PRequestInfo pRequestInfo)
 
     CHK(pRequestInfo != NULL && pRequestInfo->pAwsCredentials != NULL, STATUS_NULL_ARG);
 
+    // signAwsRequestInfo is a public api function, if service is not specified default to "kinesisvideo" so no breaking changes are introduced to the api
+    if (pRequestInfo->service == NULL)

```

Comment:
As recommended, array should be used instead of a raw pointer. In that case, the check here should be something like

if (pRequestInfo->service[0] == L'\0') { //set to kinesisvideo"} 

Ideally, this logic should be applied at the caller layer rather than here. This should simply check for it not being NULL and fail if it's null

Reply:
> Ideally, this logic should be applied at the caller layer rather than here. This should simply check for it not being NULL and fail if it's null

Yes, I agree. And I hope some thought and further work goes into this to present a well-designed api for this use-case. I think it is beyond the scope of this PR bc I do not want to break or affect the public api in any way (with the hopes that this PR is extremely minimal/simple and can be merged). I would be happy to help in a followup PR, and definitely review and adopt any PRs that aim to extend this.